src/main/java/ch/njol/skript/expressions/ExprAdvancementMessage.java

```diff
@@ -0,0 +1,99 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+import net.kyori.adventure.text.Component;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerAdvancementDoneEvent;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Objects;
+
+@Name("Advancement message")
+@Description("Sets the advancement message for on advancement: event")
+@Examples("set the advancement message to \"%event-player% did an advancement!\"")
+@Since("INSERT VERSION")
+public class ExprAdvancementMessage extends SimpleExpression<String> {
+
+    static {
+        Skript.registerExpression(ExprAdvancementMessage.class, String.class, ExpressionType.SIMPLE, "[the] advancement message");
+    }
+
+    @Override
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+        if (ScriptLoader.isCurrentEvent(PlayerAdvancementDoneEvent.class)) {
+            return true;
+        }
+        return false;
+    }
+    @Override
+    @Nullable
+    protected String[] get(Event e) {
+        return new String[] {((PlayerAdvancementDoneEvent) e).message().toString()};
```

Comment:
This method is a paper adventure message, meaning it will only be supported on 1.16.5 paper and above.
Is it possible to get the message without using adventure stuff?

Reply:
not really, when i started testing its only components

Reply:
okay, that is fine, you'll just have to make a few small changes:
- You should only register this expression if the method exists. You can she Skript#methodExists to determine this.
- You should also add the `@RequiredPlugins` annotation above `@Since`. The string can be something like `Paper 1.16.5+`

Reply:
The message method is Nullable, this code doesn't support that

Reply:
> The message method is Nullable, this code doesn't support that

Should I change anything or the code is good?

Reply:
It can still throw an NPE

Reply:
> It can still throw an NPE

Oh, what should I do? 
try { and then catch it?

Reply:
```suggestion
        Component message = ((PlayerAdvancementDoneEvent) e).message();
        if (message == null)
            return null;
        return new String[] {message.toString()};
```